build-system/tasks/e2e/selenium-webdriver-controller.js

```diff
@@ -114,10 +151,17 @@ class SeleniumWebDriverController {
    * @override
    */
   async findElementXPath(xpath) {
-    const byXpath = By.xpath(xpath);
-
-    await this.driver.wait(until.elementLocated(byXpath));
-    const webElement = await this.driver.findElement(byXpath);
+    await this.maybeInstallXpath_();
+
+    const webElement = await this.driver.wait(new Condition('', async() => {
+      try {
+        const root = await this.getRoot_();
+        const results = await this.evaluate(queryXpath, xpath, root);
```

Comment:
What about using selenium's built-in locator? https://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_By.html#By.xpath

Reply:
The built-in XPath locator doesn't work for elements inside `ShadowRoot`s, unfortunately. The `wgxpath` library added in `third_party` allows evaluating XPath queries in detached document trees as well. The comment block in `query-xpath.js` has a detailed description.